graphistry/PlotterBase.py

```diff
@@ -2269,7 +2272,41 @@ def bolt(self, driver):
         res = copy.copy(self)
         res._bolt_driver = to_bolt_driver(driver)
         return res
+    
+    def spanner_init(self: Plottable, spanner_config: Dict[str, str]) -> Plottable:
+        """
+        Initializes a SpannerGraph object with the provided configuration and connects to the instance db
+
+        spanner_config dict must contain the include the following keys, credentials_file is optional:
+            - "project_id": The GCP project ID.
+            - "instance_id": The Spanner instance ID.
+            - "database_id": The Spanner database ID.
+            - "credentials_file": json file API key for service accounts 
+
+        :param spanner_config A dictionary containing the Spanner configuration. 
+        :type (Dict[str, str])
+        :return: Plottable with a Spanner connection 
+        :rtype: Plottable
+        :raises ValueError: If any of the required keys in `spanner_config` are missing or have invalid values.
+
+        """
+        res = copy.copy(self)
+
+        project_id = spanner_config["project_id"]
+        instance_id = spanner_config["instance_id"]
+        database_id = spanner_config["database_id"]
+        credentials_file = spanner_config["credentials_file"]
 
+        # check if valid 
+        required_keys = ["project_id", "instance_id", "database_id"]
+        for key in required_keys:
+            value = spanner_config.get(key)
+            if not value:  # check for None or empty values
+                raise ValueError(f"Missing or invalid value for required Spanner configuration: '{key}'")
+
+        res._spannergraph = SpannerGraph(res, project_id, instance_id, database_id, credentials_file)
+        logger.debug("Created SpannerGraph object: {res._spannergraph}")
+        return res
```

Comment:
can we have plugin logic stay defined in the plugin, and we either just do a mixin, here, or a proxied call to the plugin?

Reply:
(we'll eventually rewrite the core here, but for now, are at least trying to get the plugins encapsulated in their definition)

Reply:
this would also let us make spanner more 'lazy' wrt import costs, so just triggered on first use of a spanner method

Reply:
ex: https://github.com/graphistry/pygraphistry/blob/bd483a973a084d6a36d82253ce6229e1f39fd579/graphistry/PlotterBase.py#L1607

```python
    def from_igraph(self,
        ig,
        node_attributes: Optional[List[str]] = None,
        edge_attributes: Optional[List[str]] = None,
        load_nodes = True, load_edges = True,
        merge_if_existing = True
    ):
        return from_igraph_base(
            self,
            ig,
            node_attributes,
            edge_attributes,
            load_nodes, load_edges,
            merge_if_existing
        )
    from_igraph.__doc__ = from_igraph_base.__doc__
```